### ISSUE TITLE
Completion of error handling

### ISSUE THREAD

**Issue by elfring** [infinisql/infinisql#12]
I have looked at a few source files for your current software. I have noticed that some checks for return codes are missing.

Would you like to add more error handling for return values from functions like the following?
- [pthread_mutex_init](http://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_mutex_init.html) ⇒ [Mbox constructor](https://github.com/infinisql/infinisql/blob/93d92966e86fbb9c50bca13ccb674670e3db5da1/infinisqld/Mbox.cc#L33)
- [pthread_mutex_lock](http://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_mutex_lock.html) ⇒ [Listener constructor](https://github.com/infinisql/infinisql/blob/93d92966e86fbb9c50bca13ccb674670e3db5da1/infinisqld/Listener.cc#L31)

**Comment by mtravis15432** [comment]
Yes, I'm interested in this type of thing. The model that I use for errors (currently) is primitive:
fprintf(logfile, "%s %i thisnthat\n", **FILE**, **LINE**);

Some stuff like failed mutex init probably justifies system exit(), but for the most part, just getting error detection in there is a good thing. I'm in favor of this.

**Comment by elfring** [comment]
Are you interested to apply [aspect-oriented software development](http://aspectc.org/)?
How do you think about to encapsulate error detection and corresponding exception handling as [a reusable aspect](http://en.wikipedia.org/wiki/Aspect_%28computer_programming%29) in your software?

**Comment by mtravis15432** [comment]
Looks promising, but I'd need to delve further into it to understand it better. As is clear, I've not put extensive effort so far into error detection & handling, though they are critical, of course, especially in an RDBMS. I'm interested in a framework to make error handling robust, and easy to incorporate into day to day efforts.
